nifi-marklogic-processors/src/main/java/org/apache/nifi/marklogic/processor/QueryMarkLogic.java

```diff
@@ -198,11 +198,19 @@ protected Collection<ValidationResult> customValidate(final ValidationContext va
      * @throws ProcessException
      */
     @Override
+    // TODO: Ask Rob how to pass the flow file to success areas.
+    // TODO: Ask Rob how to get a toString representation of the query definition that was run by the processor and save it to the attribute.
```

Comment:
You'll need to modify `newQueryBatcher` in `QueryBatcherBuilder` so that the query definition is returned as well, which will allow `QueryMarkLogic` to then call `toString()` on it (which hopefully returns something useful, I haven't tried that and thus don't know).

It's fine to modify that method, it's just a utility class. 

Note that a `Tuple` is currently being used - that's a convenience class in Java for carrying two things around. Now that we have 3 things to carry around, I think it's worth making a class for those - e.g. `QueryBatcherContext` which has fields of a DataMovementManager, QueryBatcher, and QueryDefinition.